cmd/gazelle/update-repos.go

```diff
@@ -253,7 +255,11 @@ func updateImportPaths(c *updateReposConfig, workspace *rule.File, destFile *rul
 			return nil, err
 		}
 	}
-	files := repo.MergeRules(genRules, reposByFile, destFile, kinds)
+	var emptyRules []*rule.Rule
```

Comment:
Remove this. If I understand this correctly, it would delete rules not explicitly mentioned on the command line.

I don't think a complete implementation of this is feasible right now, so just report an error in `CheckFlags` if `-prune` is set without `-from_file`. Ideally we would remove repositories that aren't needed by any libraries in the current repository or in other repositories. That requires a much more fleshed out `update-repos` in general (which unfortunately I don't have time to work on or even review right now).